functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,232 @@
+
+func teardownTestEnv(t *testing.T, testEnv *envtest.Environment) {
+	if err := testEnv.Stop(); err != nil {
+		t.Log("Error stopping test Environment\n", err)
+	}
+}
+
+func instascaleAppwrapper(namespace string) *mcadv1beta1.AppWrapper {
+	aw := &mcadv1beta1.AppWrapper{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-instascale",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"orderedinstance": "test.instance1",
+			},
+			Finalizers: []string{"instascale.codeflare.dev/finalizer"},
+		},
+		Spec: mcadv1beta1.AppWrapperSpec{
+			AggrResources: mcadv1beta1.AppWrapperResourceList{
+				GenericItems: []mcadv1beta1.AppWrapperGenericResource{
+					{
+						DesiredAvailable: 1,
+						CustomPodResources: []mcadv1beta1.CustomPodResourceTemplate{
+							{
+								Replicas: 1,
+								Requests: apiv1.ResourceList{
+									apiv1.ResourceCPU:    resource.MustParse("250m"),
+									apiv1.ResourceMemory: resource.MustParse("512Mi"),
+								},
+								Limits: apiv1.ResourceList{
+									apiv1.ResourceCPU:    resource.MustParse("1"),
+									apiv1.ResourceMemory: resource.MustParse("1G"),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return aw
+}
+
+func MachineSet(test Test, namespace string, machineSetName string) func(g Gomega) *machinev1beta1.MachineSet {
+	return func(g Gomega) *machinev1beta1.MachineSet {
+		machineset, err := test.Client().Machine().MachineV1beta1().MachineSets(namespace).Get(test.Ctx(), machineSetName, metav1.GetOptions{})
+		test.Expect(err).ToNot(HaveOccurred())
+		return machineset
+	}
+}
+
+func MachineSetReplicas(machineSet *machinev1beta1.MachineSet) *int32 {
+	return machineSet.Spec.Replicas
+}
+
+func TestReconciler(t *testing.T) {
+	// initiate setting up the EnvTest environment
+	testEnv, cfg := startEnvTest(t)
+
+	test := WithConfig(t, cfg)
+	startInstascaleController(test, testEnv, cfg) // setup client required for test to interact with kubernetes machine API
+
+	// initialize a variable with test client
+	client := test.Client()
+
+	//read machineset yaml from file `test_instascale_machineset.yml`
+	b, err := os.ReadFile("test_instascale_machineset.yml")
+	test.Expect(err).ToNot(HaveOccurred())
+
+	//deserialize kubernetes object
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	ms, _, err := decode(b, nil, nil) //decode machineset content of YAML file into kubernetes object
+	test.Expect(err).ToNot(HaveOccurred())
+	msa := ms.(*machinev1beta1.MachineSet) //asserts that decoded object is of type `*machinev1beta1.MachineSet`
+
+	//create machineset in default namespace
+	ms, err = client.Machine().MachineV1beta1().MachineSets("default").Create(test.Ctx(), msa, metav1.CreateOptions{})
+	test.Expect(err).ToNot(HaveOccurred())
+
+	//assert that the replicas in Machineset specification is 0, before appwrapper creation
+	test.Eventually(MachineSet(test, "default", "test-instascale")).Should(WithTransform(MachineSetReplicas, gstruct.PointTo(Equal(int32(0)))))
```

Comment:
Default timeout for eventually is set to 1 minute in codeflare-common.
Can you reduce it here, so the test won't wait for 1 minute before failing? IMHO 10 seconds should be enough.
Same for all other `Eventually` assertions here.